php-frontend/src/main/java/org/sonar/php/utils/SymbolUtils.java

```diff
@@ -0,0 +1,102 @@
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.declaration.NamespaceNameTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.MemberAccessTree;
+import org.sonar.plugins.php.api.tree.expression.NameIdentifierTree;
+import org.sonar.plugins.php.api.tree.expression.VariableIdentifierTree;
+
+import static org.sonar.plugins.php.api.tree.Tree.Kind.ANONYMOUS_CLASS;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.CLASS_DECLARATION;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.CLASS_MEMBER_ACCESS;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.NAMESPACE_NAME;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.NAME_IDENTIFIER;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.NEW_EXPRESSION;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.OBJECT_MEMBER_ACCESS;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.VARIABLE_IDENTIFIER;
+
+public class SymbolUtils {
+
+  private SymbolUtils() {}
+
+  public static boolean isFunctionCall(FunctionCallTree functionCallTree) {
+    return functionCallTree.callee().is(NAMESPACE_NAME) && !isNewExpressionCall(functionCallTree);
+  }
+
+  public static boolean isNewExpressionCall(FunctionCallTree functionCallTree) {
+    return functionCallTree.getParent() != null && functionCallTree.getParent().is(NEW_EXPRESSION);
+  }
+
+  public static boolean isMethodCall(FunctionCallTree functionCallTree) {
+    return functionCallTree.callee().is(OBJECT_MEMBER_ACCESS) || functionCallTree.callee().is(CLASS_MEMBER_ACCESS);
+  }
+
+  /**
+   * check whether member access of dynamic method call is resolvable as class inner call (e.g. $this->foo())
+   */
+  private static boolean isResolvableInnerObjectMemberAccess(MemberAccessTree memberAccessTree) {
+    if (memberAccessTree.member().is(NAME_IDENTIFIER)) {
+      Tree object = memberAccessTree.object();
+      return object.is(VARIABLE_IDENTIFIER) && ((VariableIdentifierTree) object).text().equals("$this");
+    }
+
+    return false;
+  }
+
+  /**
+   * check whether member access of static method call is resolvable as class inner call (e.g. self::foo(), or static::foo())
+   */
+  private static boolean isResolvableInnerClassMemberAccess(MemberAccessTree memberAccessTree) {
+    if (memberAccessTree.member().is(NAME_IDENTIFIER)) {
+      Tree object = memberAccessTree.object();
+      return ((object.is(NAMESPACE_NAME) && ((NamespaceNameTree) object).fullName().equals("self"))
+        || (object.is(NAME_IDENTIFIER) && ((NameIdentifierTree) object).text().equals("static")));
+    }
+    return false;
+  }
+
+  private static boolean isInnerClassMemberAccess(MemberAccessTree memberAccessTree) {
+    return TreeUtils.findAncestorWithKind(memberAccessTree, ImmutableSet.of(CLASS_DECLARATION, ANONYMOUS_CLASS)) != null;
+  }
+
+  public static boolean isResolvableInnerMemberAccess(MemberAccessTree memberAccessTree) {
```

Comment:
What does "Inner" mean in all those method names? In Java, an "inner class" is a class which is nested in another one.